Calculator/Calculator/ExpressionParser.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  ExpressionParser.swift
+//  Calculator
+//
+//  Created by 천승현 on 2023/01/27.
+//
+
+import Foundation
+
+enum ExpressionParser {
+    static func parse(from input: String) -> Formula {
+        var operands = CalculatorItemQueue<Double>()
+        var operators = CalculatorItemQueue<Operator>()
+        
+        let opers = input.split(whereSeparator: { $0.isNumber || $0 == "." })
+            
+        opers.forEach { oper in
+            if let operChar = oper.first, let oper = Operator.init(rawValue: operChar) {
+                operators.enqueue(oper)
+            }
+        }
+    
+        componentnsByOperators(from: input)
+            .map { $0.contains("⎼") ? ("-" + $0[$0.index(after: $0.startIndex)...]) : $0 }
+            .compactMap{ Double($0) }
+            .forEach { operands.enqueue($0) }
+        
+        return Formula(operands: operands, operators: operators)
+    }
+    
+    static private func componentnsByOperators(from input: String) -> [String] {
+        var results: [String] = [input]
+        
+        Operator.allCases.forEach { oper in
+            results = results.map { $0.split(with: oper.rawValue) }.flatMap { $0 }
+        }
```

Comment:
고차함수의 활용 좋습니다 👍
한 줄로 길게 쓰는 것 보다는 개행을 넣어주는게 가독성이 좋을 것 같아요~

역시 `oper` 축약형은 지양해주세요!
```suggestion
        Operator.allCases.forEach { oper in
            results = results.map { 
                $0.split(with: oper.rawValue) 
            }.flatMap { $0 }
        }
```